### PR TITLE
Implemented an api_token endpoint for creating service user sessions

### PR DESCRIPTION
**Pull request by cmoesgaard**

### CODE REVIEW

File: flask_saml_sso/sso.py [hunk 1]
```diff
@@ -118,6 +115,31 @@ def wrapper():
     return wrapper
 
 
+@blueprint.route('/api_token/')
```
Comment (danchr) [inline flask_saml_sso/sso.py]: `_` is somewhat unusual in paths:
```suggestion
@blueprint.route('/api-token/')
```

File: flask_saml_sso/session.py [hunk 2]
```diff
@@ -1,8 +1,31 @@
...
+LOGGED_IN = 'loggedIn'
+SAML_SESSION_INDEX = 'samlSessionIndex'
+SAML_NAME_ID = 'samlNameId'
+SAML_ATTRIBUTES = 'samlAttributes'
+SAML_SESSION_TYPE = 'samlSessionType'
+
+
+class SessionType(enum.Enum):
+    User = 1
+    Service = 2
```
Comment (danchr) [inline flask_saml_sso/session.py]: If this ends up in the database, wouldn't string keys be cleaner?

Comment (cmoesgaard) [reply flask_saml_sso/session.py]: It ends up pickled in the database, so it shouldn't make a difference.

**Comment by cmoesgaard** [comment]
> We discussed requiring a separate group for token creation. Does this PR include that?

It doesn't, no. I'll add that.

**Pull request merged**